configure

```diff
@@ -5670,14 +5679,31 @@ fi
     fi
 fi
 
+#############################################################
+# Build into shared object (pic)
+#############################################################
+
+if test ."$enable_shared" == .yes
+then
+    # compile as position independent code.
+    # -fpic is apparently faster then -fPIC, but -fPIC works always.
+    # From a SO comment (https://stackoverflow.com/a/3544211/3384414):
+    #  What's more: I did a little experiment here (on x86_64
+    #  platform), -fPIC and -fpic appears to have generated the same
+    #  code. It seems they generate a different code only on m68k,
+    #  PowerPC and SPARC.
+    # Therfore use -fPIC for now
+    CXXFLAGS+=" -fPIC"
```

Comment:
This requests position independent code in both the static and shared builds. This could lead to a performance hit for the (still default) static build according to the post [here](https://gcc.gnu.org/ml/gcc-help/2009-05/msg00259.html). 

Hopefully the person typing `--enable-shared` knows what they're doing but it might be nice to make the default target the shared library when configured with `--enable-shared` so the user has to go out of their way to generate the static library when configured with shared support?